Sources/Epoxy/Core/ButtonRow.swift

```diff
@@ -0,0 +1,203 @@
+
+/// Replace `UIImageView` in Epoxy component
+///
+/// If you want to extend, consider building your own view with
+/// the help of `ButtonRow.Style`, `ButtonRow.Content` and `ButtonRow.Behaviors`.
+public final class ButtonRow: UIButton {
+    /// Closure for touch down event.
+    private lazy var didTouchDown: ButtonClosure? = nil
+
+    /// Closure for tap event.
+    private lazy var didTap: ButtonClosure? = nil
+}
+
+// MARK: Action
+
+extension ButtonRow {
+    @objc
+    private func buttonDidTouchDown(_ button: UIButton) {
+        didTouchDown?(button)
+    }
+
+    @objc
+    private func buttonDidClick(_ button: UIButton) {
+        didTap?(button)
+    }
+}
+
+// MARK: StyledView
+
+extension ButtonRow: StyledView {
+    public struct Style: Hashable {
+        /// The tint color.
+        public let tintColor: UIColor?
+
+        /// The button type.
+        public let type: UIButton.ButtonType
+
+        /// The title style.
+        ///
+        /// The `color` property will be ignored.
+        /// Please use `ButtonRow.Content` to set the text color for different states.
+        public let titleStyle: TextRow.Style?
+
+        public init(
+            tintColor: UIColor? = nil,
+            type: UIButton.ButtonType = .system,
+            titleStyle: TextRow.Style? = nil
+        ) {
+            self.tintColor = tintColor
+            self.type = type
+            self.titleStyle = titleStyle
+        }
+    }
+
+    public convenience init(style: Style) {
+        self.init(type: style.type)
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        tintColor = style.tintColor
+
+        if let titleStyle = style.titleStyle {
+            titleLabel?.do {
+                $0.font = titleStyle.font
+                $0.textAlignment = titleStyle.alignment
+                $0.numberOfLines = titleStyle.numberOfLines
+                $0.lineBreakMode = titleStyle.lineBreakMode
+            }
+        }
+
+        addTarget(self, action: #selector(buttonDidTouchDown(_:)), for: .touchDown)
+        addTarget(self, action: #selector(buttonDidClick(_:)), for: .touchUpInside)
+    }
+}
+
+// MARK: ContentConfigurableView
+
+extension ButtonRow: ContentConfigurableView {
+    public enum Content: Equatable {
+        public struct StateContent: Equatable {
+            public let image: ImageRow.ImageType?
+            public let title: TextRow.Content?
+            public let titleColor: UIColor
+
+            public init(
+                image: ImageRow.ImageType? = nil,
+                title: TextRow.Content? = nil,
+                titleColor: ConvertibleToColor = UIColor.label
+            ) {
+                self.image = image
+                self.title = title
+                self.titleColor = titleColor.color
+            }
+        }
+
+        case normal(StateContent)
+        case disabled(StateContent)
+        case selected(StateContent)
+        case highlighted(StateContent)
+    }
```

Comment:
UIButton's `image`, `title`, and `titleColor` have different values in different states.

Considering that Epoxy will use `Style` as an identifier for reuse, some states of UIButton are not suitable to be placed in `Style`.

So here, `Content` is designed as an enum, and the state and the content in that state are set at the same time.

Reply:
In addition, normal, disabled, selected and highlighted are four common states, and other states are not considered for the time being. Please indicate if necessary